Mathlib/Geometry/Euclidean/Similarity.lean

```diff
@@ -0,0 +1,157 @@
+    exact h_not_col
+  have hangle_2 : Real.sin (∠ c a b) ≠ 0 := by
+    apply sin_ne_zero_of_not_collinear
+    rw [Set.insert_comm, Set.pair_comm]
+    exact h_not_col
+  have h_sin1 := law_sin c a b
+  have h_sin1' := law_sin c' a' b'
+  rw [← eq_div_iff_mul_eq (by positivity)] at h_sin1 h_sin1'
+  rw [← h₃, ← h₂] at h_sin1'
+  rw [h_sin1', mul_div_assoc, mul_div_assoc, mul_right_inj' hangle_1] at h_sin1
+  rw [div_eq_div_iff (by positivity) (by positivity), mul_comm] at h_sin1
+  have h_sin2 := law_sin a b c
+  have h_sin2' := law_sin a' b' c'
+  rw [← eq_div_iff_mul_eq (by positivity)] at h_sin2 h_sin2'
+  rw [← h₁, ← h₃] at h_sin2'
+  rw [h_sin2', mul_div_assoc, mul_div_assoc, mul_right_inj' hangle_2] at h_sin2
+  rw [div_eq_div_iff (by positivity) (by positivity), mul_comm] at h_sin2
+  apply Similar.reverse_of_three
+  apply Similar.comm_left
+  exact similar_of_side_side (by positivity) (by positivity) h_sin2 h_sin1.symm
+
+/-- If two triangles have proportional adjacent sides and an equal included angle, then the
+triangles are similar. -/
+theorem similar_of_side_angle_side (h_not_col : ¬ Collinear ℝ {a, b, c})
+    (h_not_col' : ¬ Collinear ℝ {a', b', c'}) (h : ∠ a b c = ∠ a' b' c')
+    (hd : dist a b * dist b' c' = dist b c * dist a' b') :
+    ![a, b, c] ∼ ![a', b', c'] := by
+  have dist_a'b' : dist a' b' ≠ 0 := by simp [ne₁₂_of_not_collinear h_not_col']
+  have dist_b'c' : dist b' c' ≠ 0 := by simp [ne₂₃_of_not_collinear h_not_col']
+  rw [← div_eq_div_iff dist_a'b' dist_b'c'] at hd
+  set k := (dist a b / dist a' b') with hk
+  have k_ne : 0 < k := by
+    rw [hk]
+    apply div_pos
+    · simp [dist_pos, ne₁₂_of_not_collinear h_not_col]
+    · simp [dist_pos, ne₁₂_of_not_collinear h_not_col']
+  have h_ab : dist a b = k * dist a' b' := by
+    rw [hk, div_mul, div_self dist_a'b', div_one]
+  have h_bc : dist b c = k * dist b' c' := by
+    rw [hd, div_mul, div_self dist_b'c', div_one]
+  have hcos := law_cos a b c
+  rw [dist_comm b _ , dist_comm b' _] at h_bc
+  rw [h_ab, h_bc] at hcos
+  field_simp at hcos
+  rw [h] at hcos
+  have hcos' := law_cos a' b' c'
+  field_simp at hcos'
+  rw [← hcos', ← mul_pow] at hcos
+  have dist_ac : 0 < dist a c := by
+    simp [dist_pos, ne₁₃_of_not_collinear h_not_col]
+  have dist_a'c' : 0 < dist a' c' := by
+    simp [dist_pos, ne₁₃_of_not_collinear h_not_col']
+  have k_dist_a'c' : 0 ≤ k * dist a' c' := by positivity
+  rw [pow_left_inj₀ (le_of_lt dist_ac) k_dist_a'c' (by norm_num)] at hcos
+  rw [dist_comm c _, dist_comm c' _] at h_bc
+  rw [dist_comm a _, dist_comm a' _] at hcos
+  rw [similar_iff_exists_pos_pairwise_dist_eq]
+  use k
+  refine ⟨k_ne, ?_⟩
+  intro i j hij
+  fin_cases i <;> fin_cases j <;> try {rw [dist_self, dist_self, mul_zero]}
+  all_goals simp; grind [dist_comm]
+
+/-- For two similar triangles, the corresponding angles are equal. -/
+theorem angle_eq_of_similar (h : ![a, b, c] ∼ ![a', b', c']) :
+    ∠ a b c = ∠ a' b' c' := by
```

Comment:
I think this should be in the `Similar` namespace to work with dot notation, so `_root_.Similar.angle_eq`.